### DISCUSSION TITLE
Attributes to pass to add function in Navigation.php

### DISCUSSION THREAD

**Discussion by fefo-p** [spatie/laravel-navigation#15] [Q&A] [answered]
Hi!
Is there a reason as to why `Section` has the ability to receive attributes but `Navigation` doesn't?

I'm trying to set the `active` property when adding the menu item.

I can do it without any problems when the item is a subitem from a menu however.

Thanks!

Fernando P.

**Comment by ivalkenburg** [comment, marked as answer]
According to the merged code the author only added the attributes parameter to the Section class and not the Navigation class (possibly forgotten). So the example that was given does not work.

I guess for now you can pass a callback which adds the the attributes

```php
Navigation::make()->add('Home', route('admin.home'), fn($s) => $s->attributes(['icon' => 'fa-solid fa-house']))
```
Maybe add macro support to the Navigation/Section class would be nice. But since this package is framework agnostic its not as simple as adding a trait from Laravel.

Edit: oh wait im dumb, its in the name. Just doesn't have the requirements in the composer.json.
